internal/handlers/pg/pjson/schema.go

```diff
@@ -125,3 +121,145 @@ func (s *schema) Unmarshal(b []byte) error {
 
 	return nil
 }
+
+// makeSchema makes schema for the given document based on its data.
+func makeSchema(td *types.Document) ([]byte, error) {
+	var buf bytes.Buffer
+
+	keys := td.Keys()
+	if len(keys) == 0 {
+		return []byte("{}"), nil
+	}
+
+	buf.WriteString(`{"p":{`)
+
+	for i, key := range keys {
+		if i > 0 {
+			buf.WriteByte(',')
+		}
+
+		var b []byte
+		var err error
+
+		if b, err = json.Marshal(key); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		buf.Write(b)
+		buf.WriteByte(':')
+
+		b, err = makeElemSchema(must.NotFail(td.Get(key)))
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		buf.Write(b)
+	}
+
+	buf.WriteString(`},"$k":`)
+
+	b, err := json.Marshal(keys)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	buf.Write(b)
+	buf.WriteByte('}')
+
+	return buf.Bytes(), nil
+}
+
+// makeElemSchema makes schema for the given element based on its data.
+func makeElemSchema(value any) ([]byte, error) {
```

Comment:
Maybe we can call it `marshalValueSchema`?

Reply:
`elem` is how schema elem data type called (see in the beginning of this file), this is why I wanted to call it `elem`. However, `elem` itself is not a good name either, so maybe it needs to be renamed too.